vehicle_maxus_t90 fails to build: missing OBDII dependency in Kconfig

## Summary

The Maxus T90 EV vehicle component fails to compile when `CONFIG_OVMS_VEHICLE_MAXT90=y` and `CONFIG_OVMS_VEHICLE_OBDII` is unset, with:

```
components/vehicle_maxus_t90/src/vehicle_maxt90.h:4:27: fatal error: vehicle_obdii.h: No such file or directory
```

## Background

The vehicle was added in #1242. The `select` was missing in that PR; the bug went unnoticed because the most common starting template (`sdkconfig.default.hw31`) sets `OBDII=y` independently.

## Root cause

`OvmsVehicleMaxt90` subclasses `OvmsVehicleOBDII` and `vehicle_maxt90.h` includes `vehicle_obdii.h`. However, the `vehicle_obdii` component's include directory and sources are gated by `CONFIG_OVMS_VEHICLE_OBDII` in `components/vehicle_obdii/component.mk`:

```make
ifdef CONFIG_OVMS_VEHICLE_OBDII
COMPONENT_ADD_INCLUDEDIRS:=src
COMPONENT_SRCDIRS:=src
...
endif
```

The MAXT90 Kconfig entry in `main/Kconfig` does not `select OVMS_VEHICLE_OBDII`, so enabling MAXT90 doesn't pull OBDII in. MAXT90 is currently the only vehicle in-tree that subclasses `OvmsVehicleOBDII`.

## Repro

From a clean checkout, with a `sdkconfig` that has `CONFIG_OVMS_VEHICLE_OBDII` unset:

```
cd vehicle/OVMS.V3
make defconfig         # MAXT90 gets enabled by default
make -j$(nproc) all    # fails in vehicle_maxt90.cpp
```

## Fix

One line in `vehicle/OVMS.V3/main/Kconfig`:

```diff
 config OVMS_VEHICLE_MAXT90
     bool "Maxus T90 EV (MT90)"
     default y
+    select OVMS_VEHICLE_OBDII
     help
       Enable to include support for Maxus T90 EV.
```

PR to follow.

--- a/vehicle/OVMS.V3/main/Kconfig
+++ b/vehicle/OVMS.V3/main/Kconfig
@@ -370,6 +370,7 @@ config OVMS_VEHICLE_NONE
 config OVMS_VEHICLE_MAXT90
     bool "Maxus T90 EV (MT90)"
     default y
+    depends on OVMS_VEHICLE_OBDII
     help
       Enable to include support for Maxus T90 EV.
 

